### PR TITLE
Upgrade to CakePHP 3.4

### PR DESCRIPTION
**Pull request by ADmad**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | maybe
| Deprecations?   | no
| Related tickets | n/a
| Documentation   | n/a
| License         | MIT

#### What's in this PR?

CakePHP dependency has been updated to 3.4

#### Why?

CakePHP 3.4 supports PSR7 so lesser code is required by this adapter. Easier maintenance :)

#### Checklist

- [ ] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [ ] Documentation pull request created (if not simply a bugfix)

### CODE REVIEW

File: src/Client.php
Comment (Nyholm) [inline src/Client.php:40]: Do we need to create a new Request if CachePHP 3.4 supports PHP7?

**Comment by Nyholm** [comment]
Thank you. I think this is a good idea. Just make sure the tests are green and I'll be happy to review and merge.

File: src/Client.php
Comment (ADmad) [reply src/Client.php:40]: Yes unfortunately Cake's HTTP client's method argument is typehinted as the concrete class `Cake\Http\Client\Request` instead of PSR7's interface. So for the time being this is required.

Comment (Nyholm) [reply src/Client.php:40]: Hm, okej. Thanks.

File: composer.json
Comment (Nyholm) [inline composer.json]: Use 3.4.5 and the test will pass. 

Versions before 3.4.5 had wrong dependencies.

**Review by Nyholm — CHANGES_REQUESTED** [review]

**Comment by Nyholm** [comment]
hm, more errors. Try bumping the versions even more then =)

**Comment by ADmad** [comment]
:tada:

File: src/Client.php
Comment (Nyholm) [inline src/Client.php]: We should use "yoda" syntax. Please revert this change.

**Review by Nyholm — COMMENTED** [review]
Awesome. There is just a code style fix. 
I think this looks good and I will merge this one if not @joelwurtz disagrees

**Comment by Nyholm** [comment]
Also. It would be great if you added a comment in the changelog for 0.2.0

**Comment by ADmad** [comment]
I'll leave it to use guys to add appropriate info to changelog :smile:

**Comment by ADmad** [comment]
@Nyholm https://dev.to/greg0ire/why-using-yoda-conditions-you-should-probably-not :)

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**
